akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -110,77 +112,175 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
 
       // we should not handle streams later than the GOAWAY told us about with lastStreamId
       private var closedAfter: Option[Int] = None
+      private var closing: Int = 0
       private var incomingStreams = mutable.Map.empty[Int, SubStream]
       private var totalOutboundWindowLeft = Http2Protocol.InitialWindowSize
       private var streamLevelWindow = Http2Protocol.InitialWindowSize
+      // TODO: make me configurable
+      private val streamTimeout = 500.milliseconds
+      private val connectionTimeout = streamTimeout.plus(streamTimeout)
 
       def lastStreamId: Int = {
         incomingStreams.lastOption.map(_._1).getOrElse(0)
```

Comment:
well we probably also need to keep a variable of the last stream, since when we close the last stream before a new one is coming, we would've get a new lastStreamId of 1, which of course would be wrong.